### PR TITLE
Improve `findAlignedLines`

### LINKED ISSUE #669: findAlignedLines sometimes returns unexpected results
In some case it has been observed that the method `findAlignedLines`used to determine which line numbers align with each other, in the set of DiffViews, return a non correct result, leading to wrong alignment.

We can see here an example of a mergeview:
Screenshot from 2023-09-05 23-40-52
The result given by `findAlignedLines`is.
[3, 3, 3, 3]
[4, 5, 4, 5]
[11, 11, 11, 12]
[18, 18, 13, 14]
There is an additionnal padding in the left editor above line 12 causing that this line is not aligned with the 2 lines 11 of central and right editors.

After proposing this manual correction for the array return by `findAlignedLines`, alignment is fixed:
[3, 3, 3, 3]
[4, 5, 4, 5]
[11, **12** 11, 12]
[18, **19**, 13, 14]
Screenshot from 2023-09-06 00-15-44

### PR DESCRIPTION
**Pull request by fcollonval**
Fixes #669

Compute correctly the alignment for a chunk in another diff view.

**Before**

image

**After**

image

### CODE REVIEW

**Comment by fcollonval** [comment]
bot please update playwright snapshots

**Comment by HaudinFlorence** [comment]
@fcollonval thanks for this PR. I have tried it on a case that was not producing correct results (the case reported in issue https://github.com/jupyter/nbdime/issues/669 that is close to the one present in this PR) and it seems to work.

File: packages/nbdime/src/common/mergeview.ts [hunk 1]
```diff
@@ -1619,53 +1629,56 @@ export class MergeView extends Panel {
...
       ? this._base.cm.defaultLineHeight
       : this._diffViews[0].remoteEditorWidget.cm.defaultLineHeight;
     if (this._aligning) {
       return;
     }
     this._aligning = true;
     // Find matching lines
-    let linesToAlign = findAlignedLines(this._diffViews);
+    const linesToAlign = findAlignedLines(this._diffViews);
+    console.log(linesToAlign)
```
Comment (vidartf) [inline packages/nbdime/src/common/mergeview.ts:1641]: debug log statement left behind

File: packages/nbdime/src/common/mergeview.ts [hunk 2]
```diff
@@ -1179,17 +1179,26 @@ function getMatchingEditLine(baseLine: number, chunks: Chunk[]): number {
...
     }
     if (chunk.baseFrom > editLine) {
+      // Remaining chunks are after the one we are interested
+      previous = chunks[i - 1];
       break;
     }
   }
-  return toMatch.baseTo;
+  // toMatch is not in chunks list, add lines delta from the last chunk
+  console.log(previous, toMatch.baseTo + (previous ? (previous.remoteTo - previous.baseTo) : 0))
```
Comment (vidartf) [inline packages/nbdime/src/common/mergeview.ts:1200]: debug log statement left behind

File: packages/nbdime/src/common/mergeview.ts [hunk 3]
```diff
@@ -1179,17 +1179,26 @@ function getMatchingEditLine(baseLine: number, chunks: Chunk[]): number {
  *
  */
 function getMatchingEditLineLC(toMatch: Chunk, chunks: Chunk[]): number {
-  let editLine = toMatch.baseFrom;
+  console.log(toMatch, chunks)
```
Comment (vidartf) [inline packages/nbdime/src/common/mergeview.ts:1182]: log statement

**Pull request merged**